x/ibc/light-clients/solomachine/types/solomachine.go

```diff
@@ -0,0 +1,47 @@
+package types
+
+import (
+	"github.com/tendermint/tendermint/crypto"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+)
+
+var _, _, _, _ codectypes.UnpackInterfacesMessage = &ClientState{}, &ConsensusState{}, &Header{}, &HeaderData{}
+
+// UnpackInterfaces implements the UnpackInterfaceMessages.UnpackInterfaces method
+func (cs *ClientState) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
+	return cs.ConsensusState.UnpackInterfaces(unpacker)
+}
+
+// UnpackInterfaces implements the UnpackInterfaceMessages.UnpackInterfaces method
+func (cs *ConsensusState) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
+	var pubKey crypto.PubKey
+	err := unpacker.UnpackAny(cs.PublicKey, &pubKey)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// UnpackInterfaces implements the UnpackInterfaceMessages.UnpackInterfaces method
+func (h *Header) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
+	var pubKey crypto.PubKey
+	err := unpacker.UnpackAny(h.NewPublicKey, &pubKey)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
the function can be a one-liner.

BTW, why do we only unpack PubKey from the whole header? Shouldn't the interface method be called `UnpackPubKey`?

Reply:
> BTW, why do we only unpack PubKey from the whole header?

Because only PubKey is an Any.

> Shouldn't the interface method be called UnpackPubKey

The `UnpackInterfaceMessages` interface is used to call recursively `UnpackInterfaces` on higher-level structs to UnpackAny their deeply nested `Any`s, so we need to keep this name.